cmd/config-manager/main.go

```diff
@@ -90,23 +117,41 @@ func readConfig(file string) (map[string]interface{}, error) {
 	return tomlMap, nil
 }
 
-func updateNRIPlugin(config map[string]interface{}) map[string]interface{} {
+func patchContainerdConfig(config map[string]interface{}) map[string]interface{} {
 	plugins, exists := config["plugins"].(map[string]interface{})
 	if !exists {
 		log.Println("Top level plugins section not found, adding it to enable NRI...")
 		plugins = make(map[string]interface{})
 		config["plugins"] = plugins
 	}
 
-	nri, exists := plugins[nriPluginKey].(map[string]interface{})
+	nri, exists := plugins[containerdNriTable].(map[string]interface{})
 	if !exists {
 		log.Println("NRI plugin section not found, adding it to enable NRI...")
 		nri = make(map[string]interface{})
-		plugins[nriPluginKey] = nri
+		plugins[containerdNriTable] = nri
+	}
+
+	nri[containerdNriKey] = false
+	log.Println("Enabled NRI in containerd...")
+	return config
+}
+
+func patchCrioConfig(config map[string]interface{}) map[string]interface{} {
```

Comment:
This would be less intrusive (wrt. losing comments in the config file, etc.) if you used a cri-o configuration drop-in snippet instead of rewriting the main configuration file:

```bash
test-node ~ $ cat /etc/crio/crio.conf.d/10-enable-nri.conf
[crio.nri]
enable_nri = true
```

Reply:
We can add a drop in file but what about the restarting the CRI-O afterwards from within the Pod?

Reply:
One option is, we mount the drop in file and from within the main.go we do the restart if the runtime is CRI-O.

Reply:
Based on the last call, I will replace the current table modification via drop in file.